src/main/scala/zio/prelude/NonEmptySet.scala

```diff
@@ -0,0 +1,207 @@
+package zio.prelude
+
+import zio.NonEmptyChunk
+
+import scala.language.implicitConversions
+
+final class NonEmptySet[A] private (val toSet: Set[A]) { self =>
+
+  /**
+   * Returns an element of this `NonEmptySet` and the remainder, which is a (possibly empty) `Set`.
+   */
+  def destruct: (A, Set[A]) = (toSet.head, toSet.tail)
```

Comment:
Maybe make this private or inline? Exposing it kind of implies that this is an efficient way to process the elements of the set which I don't think it is.

Reply:
OK, I'll make it inline. The performance will be mainly impacted by `.tail`, but that should be ["effectively constant"](https://docs.scala-lang.org/overviews/collections-2.13/performance-characteristics.html), so it should be acceptable.